os/kernel/sched/sched_releasetcb.c

```diff
@@ -234,6 +234,14 @@ int sched_releasetcb(FAR struct tcb_s *tcb, uint8_t ttype)
 		group_leave(tcb);
 #endif
 
+
+#ifdef CONFIG_BINARY_MANAGER
+	struct tcb_s *prev = tcb->bin_blink;
+	struct tcb_s *next = tcb->bin_flink;
+	if (prev) prev->bin_flink = next;
+	if (next) next->bin_blink = prev;
```

Comment:
What's the difference with dq_rem?
If this is necessary,
1. make this as a function
2. locate it in binmgr 
3. call it here

Reply:
It doesn't need queue. It is used for searching threads of same binary instead of sched_foreach.
So It just link the nodes each other.
I will make a function as you said.